pkg/mgmt/volume/volume.go

```diff
@@ -49,31 +51,113 @@ func (c *VolController) syncHandler(key string) error {
 	}
 
 	// Get the Vol resource with this namespace/name
-	Vol, err := c.VolLister.LVMVolumes(namespace).Get(name)
+	klog.Infof("Getting lvmvol object name:%s, ns:%s from cache\n", name, namespace)
+	unstructuredVol, err := c.VolLister.Namespace(namespace).Get(name)
 	if k8serror.IsNotFound(err) {
 		runtime.HandleError(fmt.Errorf("lvmvolume '%s' has been deleted", key))
 		return nil
 	}
 	if err != nil {
 		return err
 	}
-	VolCopy := Vol.DeepCopy()
+	vol := &apis.LVMVolume{}
+	err = runtimenew.DefaultUnstructuredConverter.FromUnstructured(unstructuredVol.UnstructuredContent(), &vol)
+	//err = runtime.DefaultUnstructuredConverter.FromUnstructured(Vol.UnstructuredContent(), &vol)
+	if err != nil {
+		klog.Infof("err %s, While converting unstructured obj to typed object\n", err.Error())
+	}
+	VolCopy := vol.DeepCopy()
 	err = c.syncVol(VolCopy)
 	return err
 }
 
+// addVol is the add event handler for LVMVolume
+func (c *VolController) addVol(obj interface{}) {
+	Vol, ok := c.getStructuredObject(obj)
+	if !ok {
+		runtime.HandleError(fmt.Errorf("Couldn't get Vol object %#v", obj))
+		return
+	}
+
+	if lvm.NodeID != Vol.Spec.OwnerNodeID {
+		return
+	}
+	klog.Infof("Got add event for Vol %s", Vol.Name)
+	klog.Infof("lvmvolume object to be enqueued by Add handler: %v", Vol)
+	c.enqueueVol(Vol)
+}
+
+// updateVol is the update event handler for LVMVolume
+func (c *VolController) updateVol(oldObj, newObj interface{}) {
+	newVol, ok := c.getStructuredObject(newObj)
+	if !ok {
+		runtime.HandleError(fmt.Errorf("Couldn't get Vol object %#v", newVol))
+		return
+	}
+
+	if lvm.NodeID != newVol.Spec.OwnerNodeID {
+		return
+	}
+
+	if c.isDeletionCandidate(newVol) {
+		klog.Infof("Got update event for deleted Vol %s", newVol.Name)
+		klog.Infof("Deletion timestamp for the volume from UpdateVOl: %v", newVol.ObjectMeta.DeletionTimestamp)
+		c.enqueueVol(newVol)
+	}
+}
+
+// deleteVol is the delete event handler for LVMVolume
+func (c *VolController) deleteVol(obj interface{}) {
+	Vol, ok := c.getStructuredObject(obj)
+	if !ok {
+		unstructuredObj, ok := obj.(*unstructured.Unstructured)
+		if ok {
```

Comment:
We should return some error in case of `!ok`. Right?

Reply:
Will look into it. I think logger is not required for it though..right?

Reply:
Incorporated